tests/test_gcm_push_payload.py

```diff
@@ -25,21 +37,21 @@ def test_fcm_push_payload_params(self):
 				b'{"data":{"other":"misc"},"delay_while_idle":true,'
 				b'"notification":{"body":"Hello world","title":"Push notification"},'
 				b'"registration_ids":["abc"],"time_to_live":3600}',
-				"application/json")
+				"application/json", application_id=None)
```

Comment:
1. send_bulk_message is called without setting the application_id. 
1. send_bulk_message is really send_message.
1. _cm_send_request is called by send_message with an explicit application_id=application_id, where application_id=None. So, yes it's None by default on all of these methods but, because they call each other, it is explicitly set to None by the callee.